android/src/main/kotlin/com/bbflight/background_downloader/BDPlugin.kt

```diff
@@ -833,6 +834,8 @@ class BDPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
                 var success = false
                 while (retries < 5 && !success) {
                     try {
+                        var backgroundChannel : MethodChannel? = bgChannelsByContext[contextID];
```

Comment:
The plugin field `contextID` is specific to this plugin instance -> is there a separate plugin instance for every application that uses the plugin?  If not, then `contextID` will point to the last application that registered, which may not be the one that triggered the notification. I imagine the `intent` has access to `applicationContext` so here we should then use that to look up the appropriate background channel.  

Have you been able to test notifications in your app, in various scenarios (including a different oder of starting the different apps that use the downloader?